src/marks/auto.d.ts

```diff
@@ -3,26 +3,208 @@ import type {CompoundMark, Data} from "../mark.js";
+         * Setting a reducer on **color** will automatically bin or group on
+         * both **x** and **y**.
+         */
+        reduce?: Reducer | null;
+
+        /**
+         * A valid CSS color.
+         */
+        color?: string;
+      };
+
+  /**
+   * The size encoding corresponds to the radius channel of dots; it may
+   * correspond to the length of vectors in the future.
+   */
+  size?:
+    | ChannelValue
+    | Reducer
+    | {
+        /**
+         * A string representing a column of your data, or an accessor function.
+         */
+        value?: ChannelValue;
+
+        /**
+         * A string representing a reducer, or your own reducer function.
+         * Setting a reducer on **size** will automatically bin or group on both
+         * **x** and **y**.
+         */
+        reduce?: Reducer | null;
+      };
+
+  /**
+   * The horizontal facet dimension.
+   */
   fx?: ChannelValue | {value?: ChannelValue};
+
+  /**
+   * The vertical facet dimension.
+   */
   fy?: ChannelValue | {value?: ChannelValue};
-  mark?: "dot" | "line" | "area" | "rule" | "bar";
+
+  /**
+   * A type of mark; for example, the type *bar* encompasses bar, barX, and
+   * barY. Used to guide what sort of mark **auto** should use. It should be
+   * thought of as an override; *auto* should usually do the right thing
+   * without it, and will try its best with it, but setting the mark type may
+   * lead to a nonsensical plot.
+   */
+  mark?: MarkType;
 }
 
+/**
+ * An auto options object with nothing left undefined. Automatically inferred
+ * mark type, reducers, and zero (baseline) options will be filled in; anything
+ * else left undefined will be null.
+ */
 export interface AutoSpec extends AutoOptions {
-  x: {value: ChannelValue; reduce: Reducer | null; zero?: boolean} & BinOptions;
-  y: {value: ChannelValue; reduce: Reducer | null; zero?: boolean} & BinOptions;
-  color: {value: ChannelValue; reduce: Reducer | null; color?: string};
+  /**
+   * Like auto’s **x** option, but with nothing left undefined. The reduce value
+   * may be automatically inferred to be *count* if **x** wasn’t set and a
+   * histogram will be drawn.
+   */
+  x: {value: ChannelValue; reduce: Reducer | null; zero: boolean} & BinOptions;
+
+  /**
+   * Like auto’s **y** option, but with nothing left undefined. The reduce value
+   * may be automatically inferred to be *count* if **y** wasn’t set and a
+   * histogram will be drawn.
+   */
+  y: {value: ChannelValue; reduce: Reducer | null; zero: boolean} & BinOptions;
+
+  /**
+   * Like auto’s **color** option, but with nothing left undefined.
+   */
+  color: {value: ChannelValue; reduce: Reducer | null; color: string};
+
+  /**
+   * Like auto’s **size** option, but with nothing left undefined.
+   */
   size: {value: ChannelValue; reduce: Reducer | null};
+
+  /**
+   * Like auto’s **fx** option, but with nothing left undefined.
+   */
```

Comment:
It’s better to leave these undocumented because then it will inherit the documentation from AutoOptions, which will be more informative. We can leave the “nothing left undefined” context up at the AutoSpec interface level.